### DISCUSSION TITLE
Is it possible to pause all Runs in dagster after upgrading, before we execute `dagster instance migrate`

### DISCUSSION THREAD

**Discussion by cyberjar09** [dagster-io/dagster#18059] [Q&A]
- after upgrading to a new version of dagster on prem
- I reckon its not safe to run `dagster instance migrate` while sensors or other runs are active
- one idea was to scale down daemon, webserver then run the command via a shell in the code-location
- but are there other alternatives ?
thanks

**Comment by alangenfeld** [comment]
> I reckon its not safe to run dagster instance migrate while sensors or other runs are active

I think the schema changes have all been additive, so I think the main risk is runs failing when trying to writes events due to lock contention. 

> one idea was to scale down daemon, webserver then run the command via a shell in the code-location

This is a good approach 

> but are there other alternatives ?

Not that i am aware of

**Comment by cyberjar09** [comment]
I found this in the dagster docs, seems like a cleaner approach: https://docs.dagster.io/deployment/guides/kubernetes/how-to-migrate-your-instance
